.travis.yml

```diff
@@ -13,56 +13,84 @@ notifications:
 cache:
   directories:
     - $HOME/.composer/cache
-    - $HOME/.phpbrew
+    - $HOME/.jest-cache
     - $HOME/.npm
-
-before_install:
-  - nvm install && nvm use
-  - npm install npm -g
+    - $HOME/.nvm/.cache
+    - $HOME/.phpbrew
+    - node_modules
 
 branches:
   only:
     - master
 
+before_install:
+  - nvm install
+
 jobs:
   include:
-    - stage: test
+    - name: JS unit tests
       env: WP_VERSION=latest
+      before_install:
+        - nvm install --latest-npm
+      install:
+        - npm install
       script:
-        - npm install || exit 1
-        - npm run ci || exit 1
+        - npm run lint
+        - npm run check-local-changes
+        - npm run test-unit -- --ci --maxWorkers=2 --cacheDirectory="$HOME/.jest-cache"
 
-    - stage: test
+    - name: PHP unit tests (Docker)
       env: WP_VERSION=latest DOCKER=true
       script:
         - ./bin/run-wp-unit-tests.sh
 
-    - stage: test
+    - name: PHP unit tests (PHP 5.6)
+      language: php
       php: 5.6
       env: WP_VERSION=latest
       script:
         - ./bin/run-wp-unit-tests.sh
       if: branch = master and type != "pull_request"
 
-    - stage: test
-      php: 7.1
+    - name: PHP unit tests (PHP 5.3)
       env: WP_VERSION=latest SWITCH_TO_PHP=5.3
       script:
         - ./bin/run-wp-unit-tests.sh
       if: branch = master and type != "pull_request"
 
-    - stage: test
-      php: 7.1
+    - name: PHP unit tests (PHP 5.2)
       env: WP_VERSION=latest SWITCH_TO_PHP=5.2
       script:
         - ./bin/run-wp-unit-tests.sh
 
-    - stage: test
+    - name: E2E tests (Admin with plugins) (1/2)
       env: WP_VERSION=latest POPULAR_PLUGINS=true
+      install:
+        - ./bin/setup-local-env.sh
+      script:
+        - $( npm bin )/jest --config test/e2e/jest.config.json --listTests > ~/.jest-e2e-tests
+        - npm run test-e2e -- $( awk 'NR % 2 == 0' < ~/.jest-e2e-tests )
```

Comment:
I'm not sure I understand how you split the plugin? how do you decide that a test uses a plugin or not?

Reply:
It's decided by the environment constant `POPULAR_PLUGINS=true` on line 67

Reply:
Mmm, so I misunderstood the spit :) I thought we were also counting the internal plugins (built specifically for the e2e tests)

Reply:
There's 4 jobs that run E2E tests:

1. Runs even E2E tests as an admin with popular WordPress plugins enabled
2. Runs odd E2E tests as an admin with popular WordPress plugins enabled
3. Runs even E2E tests as an author with no plugins enabled
4. Runs odd E2E tests as an author with no plugins enabled